### PR TITLE
Adding Next-Auth

### PR DESCRIPTION
**Pull request by Ketcap**
# Description

Adding Next-Auth to the project with testing of Login With GitHub

### Checklist

- [x] All the suggested authentication providers added #457 
   - [ ]  Login with email depends on email integration.

### How were these changes tested?

Add missing secrets to test login locally.

```
    GITHUB_ID: z.string().default('x'),
    GITHUB_SECRET: z.string().default('x'),

    DISCORD_ID: z.string().default('x'),
    DISCORD_SECRET: z.string().default('x'),

    TWITCH_ID: z.string().default('x'),
    TWITCH_SECRET: z.string().default('x'),

    SECRET: z.string().default('x').describe("A secret for signing the tokens and everything on next auth"),
```

#### GitHub

Callback URL: `http://localhost:3000/pasaport/callback/github`
Twitch: `http://localhost:3000/pasaport/callback/twitch`
Discord: `http://localhost:3000/pasaport/callback/discord`

### CODE REVIEW

File: packages/next-auth/lib/env.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { parseEnv, z } from "znv";
+
+export const env = parseEnv(process.env,
```
Comment (usirin) [inline packages/next-auth/lib/env.ts:3]: we gotta use the individual env variables to trigger turbo's env variable cache eslint rule.

```suggestion
export const env = parseEnv({ GITHUB_ID: process.env.GITHUB_ID, GITHUB_SECRET: process.env.GITHUB_SECRET,
```

File: packages/next-auth/src/authOptions.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    secret: env.SECRET,
+    providers: [
+      {
+        id: 'vercel-email',
+        name: 'No-Reply Kampus',
+        type: 'email',
+        server: '',
+        options: {},
+        async sendVerificationRequest({ identifier: _email, url: _url }) {
+          // Umut look over here for the email :)
```
Comment (usirin) [inline packages/next-auth/src/authOptions.ts:25]: 👋 

I thought we were gonna use the builtin [`EmailProvider`](https://next-auth.js.org/providers/email)

File: packages/next-auth/lib/env.ts (hunk 1 above)
Comment (Ketcap) [reply packages/next-auth/lib/env.ts:3]: Cizus 🤦‍♂️ I'll revert this

File: packages/next-auth/src/authOptions.ts (hunk 2 above)
Comment (Ketcap) [reply packages/next-auth/src/authOptions.ts:25]: We will include our email send here, that's how the customizable works if i understood corrrectly

Comment (Ketcap) [reply packages/next-auth/src/authOptions.ts]: Just to note on this is if we use email provider then we define the server and sender. If we want to use custom email provider this way we can call the API of the custom email provider

**Review by usirin — APPROVED** [review]

**Pull request merged**